rosbag2_transport/src/rosbag2_transport/player.cpp

```diff
@@ -158,7 +159,8 @@ void Player::prepare_publishers()
   for (const auto & topic : topics) {
     publishers_.insert(
       std::make_pair(
-        topic.name, rosbag2_transport_->create_generic_publisher(topic.name, topic.type)));
+        topic.name, rosbag2_transport_->create_generic_publisher(
+          topic.name, topic.type, Rosbag2QoS())));
   }
```

Comment:
```suggestion
    publishers_.insert({
      topic.name, 
      rosbag2_transport_->create_generic_publisher(topic.name, topic.type, Rosbag2Qos())
    });
```
I think this works. 
Also, can you replace `Rosbag2Qos()` with `Rosbag2Qos{}`?

Reply:
No problem on the `{}` (can you point me at some literature why that's better?). I'm going to leave `make_pair` because it's outside the scope of my change

Reply:
Using `{}` to invoke ctors prevents it from invoking anything but a ctor.
Its in the [isocpp core guidelines](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#es23-prefer-the--initializer-syntax)